streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -82,6 +82,37 @@ abstract class ZStream[-R, +E, +O](
   ): ZStream[R1, E1, O1] =
     ZStream.absolve(ev(self))
 
+  /**
+   * Applies an aggregator to the stream, which converts one or more elements
+   * of type `A` into elements of type `B`.
+   */
+  def aggregate[R1 <: R, E1 >: E, O1 >: O, P](sink: ZTransducer[R1, E1, O1, P]): ZStream[R1, E1, P] =
+    ZStream {
+      for {
+        pull <- self.process
+        push <- sink.push
+        done <- ZRef.makeManaged(false)
+        run = {
+          def go: ZIO[R1, Option[E1], Chunk[P]] = done.get.flatMap {
+            if (_)
+              push(None).repeat(Schedule.doWhile(_.isEmpty))
```

Comment:
What do you think of this @iravid ?

Reply:
It’s fine. A direct loop would be more performant but this should probably be an uncommon code path.